src/full_coverage_path_planner.cpp

```diff
@@ -210,15 +211,18 @@ namespace full_coverage_path_planner
     }
 
     // Save map origin and scaling
-    tile_size_ = nodeSize * cpp_grid_.info.resolution; // Size of a tile in meters
-    grid_origin_.x = cpp_grid_.info.origin.position.x; // x-origin in meters
-    grid_origin_.y = cpp_grid_.info.origin.position.y; // y-origin in meters
+    tile_size_ = nodeSize * cpp_costmap->getResolution(); // Size of a tile in meters
+    double originX, originY;
+    cpp_costmap->mapToWorld(0, 0, originX, originY);
+    grid_origin_.x = originX;
+    grid_origin_.y = originY;
```

Comment:
Why the intermediate step?

```suggestion
    cpp_costmap->mapToWorld(0, 0, grid_origin_.x, grid_origin_.y);
```

Reply:
there was a float / double issue. just change the type of object. There are more places were float is used. In the type of processors we use, should we consider floats at all?

Reply:
I would just go for doubles everywhere. Doesn't have much impact on an x64 machine.